resolwe_bio/processes/clustering/find_similar.py

```diff
@@ -0,0 +1,265 @@
+                self.error(
+                    "Input samples are of different Expression types: "
+                    f"{exp.exp_type} and {inputs.expressions[0].exp_type}."
+                )
+            if exp.feature_type != inputs.expressions[0].feature_type:
+                self.error(
+                    "Input samples are of different Feature type: "
+                    f"{exp.feature_type} and {inputs.expressions[0].feature_type}."
+                )
+
+        labels, positions = self.get_data_info(inputs.expressions)
+
+        if len(labels) == 1:
+            self.error(
+                "Only one time point was provided. At least two time "
+                "points are required."
+            )
+
+        expressions, excluded = join_expressions(positions, labels)
+        if len(expressions.index) < 2:
+            self.warning("At least two genes shared across all samples are required.")
+
+        if excluded:
+            suffix = "" if len(excluded) <= 3 else ", ..."
+            excluded_genes = ", ".join(excluded[:3])
+            self.warning(
+                "Genes not present in all of the selected samples are "
+                f"excluded from the analysis. Excluded {len(excluded)} "
+                f"of them ({excluded_genes + suffix})."
+            )
+
+        if inputs.gene not in expressions.index:
+            self.error(
+                "Selected query gene was not found. Please make sure "
+                "the selected gene name can be found in all expression "
+                "time courses."
+            )
+
+        expressions, removed = remove_const_genes(expressions)
+
+        rows = expressions.index
+        if len(rows) < 2:
+            self.error(
+                "There are less than two genes with non-constant "
+                "expression across time points. Distances can not be "
+                "computed."
+            )
+
+        suffix = "" if len(removed) <= 3 else ", ..."
+        if removed:
+            removed_genes = ", ".join(removed[:3])
+            self.warning(
+                f"{len(removed)} genes ({removed_genes+suffix}) have "
+                "constant expression across time points. Those genes "
+                "are excluded from the computation of hierarchical "
+                "clustering of genes."
+            )
+            if inputs.gene in removed:
+                self.error(
+                    f"Query gene ({inputs.gene}) has constant "
+                    "expression and was removed. Distances can not be "
+                    "computed."
+                )
+
+        distance_map = {"pearson": calculate_pearson, "spearman": calculate_spearman}
+        distance_func = distance_map[inputs.distance]
+        selected_etc = expressions.loc[inputs.gene].tolist()
+        expressions = expressions.drop(inputs.gene)
+
+        distances = np.array(
+            [distance_func(selected_etc, etc) for etc in expressions.as_matrix()]
```

Comment:
When we support running multithreaded interactive processes, we'll be able to parallelize this list comprehension to speed up the computation. But for now this is OK.